tests/integration/targets/rds_instance_snapshot/tasks/main.yml

```
@@ -0,0 +1,363 @@
+      security_token: "{{ security_token | default(omit) }}"
+  collections:
+    - community.aws
+    - amazon.aws
+
+  block:
+  - name: Create a source mariadb instance
+    rds_instance:
+      id: "{{ instance_id }}"
+      state: present
+      engine: "{{ engine}}"
+      engine_version: "{{ mariadb_engine_version }}"
+      allow_major_version_upgrade: true
+      username: "{{ username }}"
+      password: "{{ password }}"
+      db_instance_class: "{{ db_instance_class }}"
+      allocated_storage: "{{ allocated_storage }}"
+    register: _result_create_instance
+
+  - assert:
+      that:
+      - _result_create_instance.changed
+      - _result_create_instance.db_instance_identifier == "{{ instance_id }}"
+  
+  - name: Get all RDS snapshots for the existing instance
+    rds_snapshot_info:
+      db_instance_identifier: "{{ instance_id }}"
+    register: _result_instance_snapshot_info
+  
+  - assert:
+      that:
+      - _result_instance_snapshot_info is successful
+      - _result_instance_snapshot_info.snapshots | length == 1
+
+  - name: Take a snapshot of the existing RDS instance (CHECK_MODE)
+    rds_instance_snapshot:
+      state: present
+      db_instance_identifier: "{{ instance_id }}"
+      db_snapshot_identifier: "{{ snapshot_id }}"
+    check_mode: yes
+    register: _result_instance_snapshot
+
+  - assert:
+      that:
+      - _result_instance_snapshot.changed
+  
+  - name: Take a snapshot of the existing RDS instance
+    rds_instance_snapshot:
+      state: present
+      db_instance_identifier: "{{ instance_id }}"
+      db_snapshot_identifier: "{{ snapshot_id }}"
+      wait: true
+    register: _result_instance_snapshot
+
+  - assert:
+      that:
+      - _result_instance_snapshot.changed
+      - "'allocated_storage' in _result_instance_snapshot"
+      - "'availability_zone' in _result_instance_snapshot"
+      - "'instance_create_time' in _result_instance_snapshot"
+      - "'db_instance_identifier' in _result_instance_snapshot"
+      - _result_instance_snapshot.db_instance_identifier == "{{ instance_id }}"
+      - "'db_snapshot_identifier' in _result_instance_snapshot"
+      - _result_instance_snapshot.db_snapshot_identifier == "{{ snapshot_id }}"
+      - "'db_snapshot_arn' in _result_instance_snapshot"
+      - "'dbi_resource_id' in _result_instance_snapshot"
+      - "'encrypted' in _result_instance_snapshot"
+      - "'engine' in _result_instance_snapshot"
+      - _result_instance_snapshot.engine == "{{ engine }}"
+      - "'engine_version' in _result_instance_snapshot"
+      - _result_instance_snapshot.engine_version == "{{ mariadb_engine_version }}"
+      - "'iam_database_authentication_enabled' in _result_instance_snapshot"
+      - "'license_model' in _result_instance_snapshot"
+      - "'master_username' in _result_instance_snapshot"
+      - _result_instance_snapshot.master_username == "{{ username }}"
+      - "'original_snapshot_create_time' in _result_instance_snapshot"
```

```suggestion
      - "'snapshot_create_time' in _result_instance_snapshot"
```
I think these assertions should be like this, the tests pass if I change this value throughout the file (with the terminator changes)